File: acmicpc/단계별로 풀어보기/1차원 배열 사용하기/8958.c
#include <stdio.h>


int main() {
	int T;
	int count;
	char arr[80] = "";
	int sum;
	scanf("%d", &T);
	while (T) {
		T--;
		scanf("%s", &arr);
		count = 0;
		sum = 0;
		for (int i = 0; i < 79; i++) {
			if (arr[i] == '\0')
				break;
			else if (arr[i] == 'O') {
				count++;
				sum += count;
			}
			else if (arr[i] == 'X') {
				count = 0;
			}
		}
		printf("%d\n", sum);
	}
}

File: acmicpc/단계별로 풀어보기/1차원 배열 사용하기/8958.java
//OX 퀴즈

// 처음 제출 [메모리 : 13752KB 시간 : 104ms]
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Java_8958 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int TestCase = Integer.parseInt(br.readLine());
        for(int i=0; i<TestCase; i++){
            int count=1;
            int sum=0;
            String[] Input = br.readLine().split("");
            for(int j=0 ; j < Input.length; j++){
                if(Input[j].equals("O")){
                    sum+=count;
                    count++;
                } else
                    count = 1;
            }
            System.out.println(sum);
        }
    }
}
